### PR TITLE
Added support for setting column wrap and ellipsis behavior

### PR DESCRIPTION
**Pull request by petrixh**
Added support for configuring how the grid will handle text that overflows. There are two properties that can be set: 

ellipsis - makes all columns use ellipsis
wrap - makes all columns use eager wrapping (word-wrap: break-word, instead of the default "normal")

The default behavior should be the same as before, i.e. using clipping and wrapping on whitespaces and "-". 

While these seem to work, in IE11 they cannot be toggled, so just add class="ellipsis" or class="wrap" to <px-data-grid> in order to have the correct behavior. Could not figure out a way to make IE refresh itself when css classes are added/remoed, but just adding the class to the element worked. 

Since there are effectively 4 column renderers: 
default
string
date
number

there are css variables for each of them: 

 --px-data-grid-cell-COLTYPE-column-ATTRIBUTE

(see the renderers for details, or px-data-grid.scss for examples)

A couple of changes that had to be made: 
- Grid rows no longer have a fixed height when rendering (it looked bad)
- on edit, save and cancel the grid will refresh the cache to recalc the row heights (is there a more lightweight way?) 
- when going to edit mode the row height will increase/decrease to fit the editors

I'd ask for a bit of testing during the review process if possible. I've ran through the tests, tried to compare against the old implementation. I added toggles to the "custom columns" demo since all the column generators are available there. 

Fixes: #310 
Fixes: #303 
Fixes: #456

Might also help with 552
Probably also addresses 394

Replaces pull request 549 as the branch was over 40 commits behind and merging made everything messy...

### CODE REVIEW

File: px-data-grid.html [hunk 1]
```diff
@@ -1648,6 +1675,28 @@
...
         }
 
+        _ellipsisModeChanged(ellipsisMode) {
+          if (ellipsisMode) {
+            this.classList.add('ellipsis');
+          } else {
+            this.classList.remove('ellipsis');
+          }
+          if (this._vaadinGrid) {
+            this._vaadinGrid.clearCache();
```
Comment (alump) [inline px-data-grid.html]: notifyResize

File: px-data-grid.html [hunk 2]
```diff
@@ -1648,6 +1675,28 @@
...
+        }
+
+        _wrapModeChanged(wrapMode) {
+          if (wrapMode) {
+            this.classList.add('wrap');
+          } else {
+            this.classList.remove('wrap');
+          }
+          if (this._vaadinGrid) {
+            this._vaadinGrid.clearCache();
```
Comment (alump) [inline px-data-grid.html]: notifyResize

File: px-data-grid.html [hunk 3]
```diff
@@ -1442,6 +1467,8 @@
             }
 
             this._editingItem = null;
+            // To refresh row height after edit...
+            this._vaadinGrid.clearCache();
```
Comment (alump) [inline px-data-grid.html]: notifyResize

File: px-data-grid.html [hunk 4]
```diff
@@ -1407,11 +1427,16 @@
...
+
+          // To refresh row height after edit...
+          this._vaadinGrid.clearCache();
         }
 
         _cancelEdit(event) {
           this._editingRenderers.forEach((renderer) => renderer.restoreInitial());
           this._editingItem = null;
+          // To refresh row height after edit...
+          this._vaadinGrid.clearCache();
```
Comment (alump) [inline px-data-grid.html]: notifyResize

File: px-data-grid.html [hunk 5]
```diff
@@ -1407,11 +1427,16 @@
           // To prevent selections
           event.stopPropagation();
           this._editingItem = event.details.item;
+
+          // To refresh row height after edit...
+          this._vaadinGrid.clearCache();
```
Comment (alump) [inline px-data-grid.html]: notifyResize

**Comment by petrixh** [comment]
Changed the references to clearCache() to notifyResize() seems to work well. 

Based on Vladimirs comment, Sami was in contact with the customer. The customer actually wants the cells to wrap even when there are no white-spaces or dashes available and they suggested using "word-break: break-all". Also, in the spirit of the original tickets, I've removed the "wrap" style and toggle. By default wrapping is on and even though the customer suggested "break-all" for word-break, I decided to go with "break-word" instead. It will first try to pass the word to the next line making things like names wrap in a more sensible manner and only after that will it start breaking the word itself. 

@alump @limonte if you have a chance to take another look at this it would be appreciated, the case that Vladimir pointed out should now be fixed so that everything wraps, including the link URL.

**Comment by alump** [comment]
https://github.com/vaadin/px-data-grid/issues/456

File: sass/px-data-grid.scss [hunk 6]
```diff
@@ -20,10 +77,16 @@ px-data-grid-cell-content-wrapper {
   padding-left: var(--px-data-grid-padding-left);
   padding-right: var(--px-data-grid-padding-right);
   min-height: 100%;
-  display: flex;
+  display: var(--px-data-grid-deault-content-wrapper-display, block);
```
Comment (toshegg) [inline sass/px-data-grid.scss]: Typo, `deault`.

File: sass/px-data-grid.scss [hunk 7]
```diff
@@ -8,6 +8,63 @@
   --px-data-grid-padding-bottom: 12px;
   --px-data-grid-padding-left: 12px;
   --px-data-grid-padding-right: 12px;
+
+}
+
+
+/* Default behavior, pre- ellipsis and intentional wrapping:
+{
+//Make all wrap
```
Comment (toshegg) [inline sass/px-data-grid.scss]: Please, use space:
`// foobar` instead of `//foobar`

File: sass/px-data-grid.scss [hunk 8]
```diff
@@ -8,6 +8,63 @@
...
+  //Defalut columns (without custom renderers)
```
Comment (toshegg) [inline sass/px-data-grid.scss]: Please, use space:
`// foobar` instead of `//foobar`

File: sass/px-data-grid.scss [hunk 9]
```diff
@@ -8,6 +8,63 @@
...
+  --px-data-grid-cell-date-column-word-wrap: normal;
+  --px-data-grid-cell-date-column-overflow: visible;
+  --px-data-grid-cell-date-column-white-space: normal;
+  --px-data-grid-cell-date-column-word-break: normal;
+  --px-data-grid-cell-date-column-text-overflow: clip;
+} */
+
+
+:host(.ellipsis){
+  //Make all columns use ellipsis
```
Comment (toshegg) [inline sass/px-data-grid.scss]: Please, use space:
`// foobar` instead of `//foobar`

File: sass/px-data-grid.scss [hunk 10]
```diff
@@ -8,6 +8,63 @@
...
+
+  //Default columns (without custom renderers)
```
Comment (toshegg) [inline sass/px-data-grid.scss]: Please, use space:
`// foobar` instead of `//foobar`

File: sass/px-data-grid.scss [hunk 11]
```diff
@@ -8,6 +8,63 @@
...
+  --px-data-grid-cell-default-column-overflow: hidden;
+  --px-data-grid-cell-default-column-white-space: nowrap;
+  --px-data-grid-cell-default-column-text-overflow: ellipsis;
+  --px-data-grid-deault-content-wrapper-display: block;
+
+  //Specific column generator types
```
Comment (toshegg) [inline sass/px-data-grid.scss]: Please, use space:
`// foobar` instead of `//foobar`

File: px-data-grid-string-renderer.html [hunk 12]
```diff
@@ -4,6 +4,31 @@
...
+Custom property | Description
+'--px-data-grid-cell-string-column-overflow' | controls the overflow behavior, visible, hidden
+'--px-data-grid-cell-string-column-white-space' | controls how white-spaces are handled, normal, nowrap
+'--px-data-grid-cell-string-column-word-wrap' | controls how word-wrapping is handled normal, break-word
+'--px-data-grid-cell-string-column-text-overflow' | controls how text-overflow is handled, clip, ellipsis
+'--px-data-grid-cell-string-column-editor-align-items | control editor alignment when in edit mode, center, flex-end, flex-start'
+
+Examples:
+Make string-renderer columns wrap the content:
+--px-data-grid-cell-string-column-overflow: hidden;
```
Comment (toshegg) [inline px-data-grid-string-renderer.html]: IMO, <style> wrapper is needed here, i.e:

Make string-renderer columns wrap the content:
```
<style>
 px-data-grid {
  --px-data-grid-cell-string-column-overflow: hidden;
  --px-data-grid-cell-string-column-word-wrap: break-word;
 }
</style>
```

File: px-data-grid-string-renderer.html [hunk 13]
```diff
@@ -4,6 +4,31 @@
...
+--px-data-grid-cell-string-column-word-wrap: break-word;
+
+Make string-renderer columns use ellipsis overflow:
```
Comment (toshegg) [inline px-data-grid-string-renderer.html]: IMO, `<style>` wrapper is also needed here, see above.

File: px-data-grid-date-renderer.html [hunk 14]
```diff
@@ -5,6 +5,30 @@
...
+
+Custom property | Description
+'--px-data-grid-cell-date-column-overflow' | controls the overflow behavior, visible, hidden
+'--px-data-grid-cell-date-column-white-space' | controls how white-spaces are handled, normal, nowrap
+'--px-data-grid-cell-date-column-word-wrap' | controls how word-wrapping is handled normal, break-word
+'--px-data-grid-cell-date-column-text-overflow' | controls how text-overflow is handled, clip, ellipsis
+'--px-data-grid-cell-date-column-editor-align-items | control editor alignment when in edit mode, center, flex-end, flex-start'
+
+Examples:
+Make date-renderer columns wrap the content:
```
Comment (toshegg) [inline px-data-grid-date-renderer.html]: IMO, `<style>` wrapper is also needed here, see below.

File: px-data-grid-date-renderer.html [hunk 15]
```diff
@@ -5,6 +5,30 @@
...
+--px-data-grid-cell-date-column-overflow: hidden;
+--px-data-grid-cell-date-column-word-wrap: break-word;
+
+Make date-renderer columns use ellipsis overflow:
```
Comment (toshegg) [inline px-data-grid-date-renderer.html]: IMO, `<style>` wrapper is also needed here, see below.

File: px-data-grid-number-renderer.html [hunk 16]
```diff
@@ -4,6 +4,30 @@
...
+
+Custom property | Description
+'--px-data-grid-cell-number-column-overflow' | controls the overflow behavior, visible, hidden
+'--px-data-grid-cell-number-column-white-space' | controls how white-spaces are handled, normal, nowrap
+'--px-data-grid-cell-number-column-word-wrap' | controls how word-wrapping is handled normal, break-word
+'--px-data-grid-cell-number-column-text-overflow' | controls how text-overflow is handled, clip, ellipsis
+'--px-data-grid-cell-number-column-editor-align-items | control editor alignment when in edit mode, center, flex-end, flex-start'
+
+Examples:
+Make number-renderer columns wrap the content:
```
Comment (toshegg) [inline px-data-grid-number-renderer.html]: IMO, `<style>` wrapper is also needed here, see below.

File: px-data-grid-number-renderer.html [hunk 17]
```diff
@@ -4,6 +4,30 @@
...
+--px-data-grid-cell-number-column-overflow: hidden;
+--px-data-grid-cell-number-column-word-wrap: break-word;
+
+Make number-renderer columns use ellipsis overflow:
```
Comment (toshegg) [inline px-data-grid-number-renderer.html]: IMO, `<style>` wrapper is also needed here, see below.

File: px-data-grid-date-renderer.html (hunk 15 above)
Comment (petrixh) [reply px-data-grid-date-renderer.html]: This is just the documentation of the css-variables, the syntax is copied from here https://github.com/PredixDev/px-dropdown/blob/master/px-dropdown.html under the style-section.

File: px-data-grid-number-renderer.html (hunk 16 above)
Comment (petrixh) [reply px-data-grid-number-renderer.html]: This is just the documentation of the css-variables, the syntax is copied from here https://github.com/PredixDev/px-dropdown/blob/master/px-dropdown.html under the style-section.

File: px-data-grid-number-renderer.html (hunk 17 above)
Comment (petrixh) [reply px-data-grid-number-renderer.html]: This is just the documentation of the css-variables, the syntax is copied from here https://github.com/PredixDev/px-dropdown/blob/master/px-dropdown.html under the style-section.

File: px-data-grid-string-renderer.html (hunk 12 above)
Comment (petrixh) [reply px-data-grid-string-renderer.html]: This is just the documentation of the css-variables, the syntax is copied from here https://github.com/PredixDev/px-dropdown/blob/master/px-dropdown.html under the style-section.

**Pull request merged**